OpenRA.Game/Exts.cs

```diff
@@ -357,6 +357,11 @@ public static ulong ISqrt(ulong number, ISqrtRoundMode round = ISqrtRoundMode.Fl
 			return root;
 		}
 
+		public static int MultiplyBySqrtTwo(short number)
+		{
+			return number * 46341 / 32768;
```

Comment:
Is there a reason for `46341 / 32768` specifically? This doesn't seem much more precise than `577 / 408` for example. (But is ok I guess.)

Reply:
32768 was just the most obvious number to guarantee as much accuracy as possible for all possible values - since it's the biggest possible `short`. You can get away with smaller constants for sure, but since I'm not sure that's of any benefit, I didn't want to faff around determining the smallest acceptable value.